src/proto/cohort.proto

```diff
@@ -0,0 +1,56 @@
+// API for the cohorts to process transactions from the coordinator.
+
+syntax = "proto3";
+
+package cohort;
+
+import "src/proto/common.proto";
+
+message PrepareTransactionRequest {
+  // Identifier for transaction within cohorts/coordinators. Globally unique
+  // across all requests for all clients. Client can use this to request status
+  // updates and to check the blockchain themselves.
+  string transaction_id = 1;
+  // All operations should have the namespace of the current cohort. Any
+  // operations for other namespaces can be ignored.
+  common.Transaction transaction = 2;
+  // Should be the final config updated by the coordinator.
+  common.TransactionConfig config = 3;
+}
+
+message PrepareTransactionResponse {
+  oneof status {
+    common.PendingResponse pending_response = 1;
```

Comment:
Can you explain more on what are these two status? How is it different from a common status: ok, error, etc?

Reply:
The pending response is the normal response here. I guess the different abort reasons correspond to different error statuses:
* ABORT_REASON_PRESUMED_ABORT_TIMESTAMP_REACHED can be represented by DEADLINE_EXCEEDED
* ABORT_REASON_RESOURCE_LOCKED can be represented by UNAVAILABLE or ABORTED
* ABORT_REASON_OPERATION_FOR_NON_EXISTENT_VALUE can be represented by NOT_FOUND
* ABORT_REASON_RELATIVE_VALUE_INVALID_TYPE can be represented by INVALID_ARGUMENT or maybe FAILED_PRECONDITION.

I think it's fine to return either the gRPC status codes with good error messages or the enum errors.
WDYT?

Reply:
I see. We probably want to decide how much work a cohort to do before returning this call. It seems the cohort will block a bit for checking the lock/timestamp/value/etc (and the coordinator might need to wait if it's single threaded).

An alternative is: cohort takes the transaction and return 'yes' without any checks. If any of the lock/timestamp/etc issue happens, cohort simply vote 'no' to the blockchain later. This might be simpler and more efficient. WDYT?

Reply:
SGTM, this can be an empty response for now (indicating that it was received).